### PR TITLE
Add tabular_role_mapping_aws resource

### PR DESCRIPTION
**Pull request by dharrisio**

### CODE REVIEW

File: internal/provider/role_mapping_aws.go [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+			},
+			"name": schema.StringAttribute{
+				Description: "Service account name",
+				Required:    true,
+			},
+			"role_id": schema.StringAttribute{
+				Description: "Role ID",
+				Required:    true,
+			},
+			"credential_key": schema.StringAttribute{
```
Comment (maxwell-gobel) [inline internal/provider/role_mapping_aws.go:68]: Do we want to expose/is there any value to exposing that these are actually credentials under the covers?

**Review by maxwell-gobel — APPROVED** [review]
Implementation looks good; not sure we want to include credential key/secret.

File: internal/provider/role_mapping_aws.go (hunk 1 above)
Comment (dharrisio) [reply internal/provider/role_mapping_aws.go:68]: It's probably not necessary in this case. The credential key is a good choice for the `id` attribute, but it probably doesn't need to be accessed outside of that.

File: internal/provider/role_mapping_aws.go [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+	r.client = req.ProviderData.(*util.Client)
+}
+
+func (r *roleMappingAWSResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_role_mapping_aws"
+}
+
+func (r *roleMappingAWSResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Tabular Service Account",
```
Comment (smikloska) [inline internal/provider/role_mapping_aws.go:49]:
```suggestion
		Description: "AWS Role Mapping Resource",
```

File: internal/provider/role_mapping_aws.go [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+	}
+
+	r.client = req.ProviderData.(*util.Client)
+}
+
+func (r *roleMappingAWSResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_role_mapping_aws"
+}
+
+func (r *roleMappingAWSResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
```
Comment (smikloska) [inline internal/provider/role_mapping_aws.go:47]: very much a nit:
```suggestion
func (r *AWSRoleMappingResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
```

To match the AWS naming of AWSIAMPolicyDataSource? or should that one be switched to `IAMPolicyAWSDataSource`. Neutral on which ever, but would be nice to consistent in our naming when specifying it is cloud specifc

File: internal/provider/role_mapping_aws.go [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "Credential ID",
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"name": schema.StringAttribute{
+				Description: "Service account name",
```
Comment (smikloska) [inline internal/provider/role_mapping_aws.go:59]: Similar to above 
```suggestion
				Description: "AWS Role Mapping Name",
```

File: internal/provider/role_mapping_aws.go [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	credentialKey := state.Id.ValueString()
+	roleMappingAWS, _, err := r.client.V2.DefaultAPI.GetCredential(ctx, *r.client.OrganizationId, credentialKey).Execute()
+	if err != nil {
+		resp.Diagnostics.AddError("Unable to read service account", "Unable to read service account "+err.Error())
```
Comment (smikloska) [inline internal/provider/role_mapping_aws.go:96]:
```suggestion
		resp.Diagnostics.AddError("Unable to read AWS role mapping", "Unable to read service account "+err.Error())
```

**Comment by smikloska** [comment]
Code wise looks good 🎉 , just nits on naming and comment / error message copy paste from service account PR

**Pull request merged**